src/Client.php

```diff
@@ -58,35 +48,28 @@ class Client implements ClientInterface
     private $transport;
 
     /**
-     * @var IntegrationInterface[] The stack of integrations
-     */
-    private $integrations;
-
-    /**
-     * @var SerializerInterface The serializer
+     * @var EventFactory The factory to create {@see Event} from raw data
      */
-    private $serializer;
+    private $eventFactory;
 
     /**
-     * @var RepresentationSerializerInterface The representation serializer
+     * @var IntegrationInterface[] The stack of integrations
      */
-    private $representationSerializer;
+    private $integrations;
 
     /**
      * Constructor.
      *
-     * @param Options                           $options                  The client configuration
-     * @param TransportInterface                $transport                The transport
-     * @param SerializerInterface               $serializer               The serializer used for events
-     * @param RepresentationSerializerInterface $representationSerializer The representation serializer to be used with stacktrace frames
+     * @param Options            $options      The client configuration
+     * @param TransportInterface $transport    The transport
+     * @param EventFactory       $eventFactory The factory for events
      */
-    public function __construct(Options $options, TransportInterface $transport, SerializerInterface $serializer, RepresentationSerializerInterface $representationSerializer)
+    public function __construct(Options $options, TransportInterface $transport, EventFactory $eventFactory)
```

Comment:
I don't like at all that we're passing to the client a dependency that can't be mocked because it's `final` and not has an interface

Reply:
I've avoided making an interface for that because I wouldn't like the end user extending that. The factory in itself does just instantiation and population of the event, it shouldn't be an issue not mocking it.

Reply:
My thinking is that this is a dependency, so it should be mockable (you have otherwise to mock all its dependencies). Also a class should be `final` when it implements an interface or is a VO